### PR TITLE
Standardize error structure in json responses. 

### LINKED ISSUE #3666: Standardize Error Message Schema in API
## Tell us about the problem you're trying to solve
* We should create a base schema for error messages from the API. Right now the UI has to just know that different errors are formatted differently which is hard for it to handle and not very predictable

## Describe the solution you’d like
Doesn't need to be rocket science here. might just be:

```
{ message: "foo is not a bar" }
```

and then evolve with any additional _optional_ parameters (e.g. links to docs or what not) as we go.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200368060087111/1200393103356029) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by airbyte-jenny**
Standardize error structure in json responses. Use named types for raising exceptions instead of http status codes peppered throughout the code.

## What
Various exception mappers created varying json structs for returning exception info.  Some provided message and details, while others only provided message.  Message was sometimes the details and other times just a title. HTTP status codes were determined in many places in the code, causing code repetition.  Resulting errors were unpredictable for the UI to parse and display.  Lack of consistency also made it hard to add tracking details later like request id.

## How
KnownException has been standardized as the abstract parent of custom exceptions for various scenarios, and returns an ExceptionInfo in the json api response.  The API is updated to support this, so the UI can rely on it.

Code now throws more specific named exceptions instead of needing to know an http status code, and status code is determined by the concrete child exception.   This also adds stack trace lines for exception detail, and, if present, root cause exception, so that the UI can optionally show these where appropriate.

## Recommended reading order
1. `airbyte-api/src/main/openapi/config.yaml` (end of file, data structure for ExceptionInfo)
2. `airbyte-server/src/main/java/io/airbyte/server/errors/KnownExceptionMapper.java`
3. `airbyte-server/src/main/java/io/airbyte/server/errors/KnownException.java`
4. `airbyte-server/src/main/java/io/airbyte/server/errors/ValueConflictException.java`
5.  `airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java`

Closes https://github.com/airbytehq/airbyte/issues/3666

### CODE REVIEW

File: airbyte-api/src/main/openapi/config.yaml
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:1166]: should we do the same thing we did for the 422s to the 404s as well? can't remember if we said were going to hold off for some reason.

File: airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java
Comment (cgardens) [inline airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java:555]: seeing these various exceptions that extend `KnownException` next to each other, I'm wondering if there's any naming convention we should adopt so it's clearer what these exceptions are doing. Because they're now not obviously known exceptions its hard to get the queue that these will be handled beautifully in the API. e.g. add KnownException (or some abbreviation, or some other naming convention entirely).

**Review by cgardens — APPROVED** [review]
A couple suggestions. LMK if you'd like to go over any of them.

Looking good! Now we can tell consumers we provide a sane, consistent interface when returning 400s/500s

File: airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java
Comment (airbyte-jenny) [reply airbyte-server/src/main/java/io/airbyte/server/apis/ConfigurationApi.java:555]: Yeah, I agree.  They should all include KnownException in the name, so that the hierarchy is more obvious, and also so they show up as autocomplete suggestions during IDE searches.

File: airbyte-api/src/main/openapi/config.yaml
Comment (airbyte-jenny) [reply airbyte-api/src/main/openapi/config.yaml:1166]: That's a good idea.  The bit I was holding off on was trying to figure out if "id" ought to be its own attribute; I think it can be.  Every scenario has one.  I will include this change as well.

File: airbyte-api/src/main/openapi/config.yaml
Comment (cgardens) [inline airbyte-api/src/main/openapi/config.yaml:2670]: bummer that we have to lose the inheritance. i think it would be worthwhile leaving a comment explaining what the "required" fields of all exceptions are. maybe also worth mentioning why we are duplicating attributes as opposed to using inheritance.

File: airbyte-server/src/main/java/io/airbyte/server/errors/IdNotFoundKnownException.java
Comment (cgardens) [inline airbyte-server/src/main/java/io/airbyte/server/errors/IdNotFoundKnownException.java:59]: i don't think you should have to use `this` in front of any of these method calls. welcome to keep it if you think it is clearer.

**Review by cgardens — APPROVED** [review]
Looks good. 2 last thoughts but once you look at them feel free to merge it!

File: airbyte-server/src/main/java/io/airbyte/server/errors/NotFoundExceptionMapper.java
Comment (davinchia) [inline airbyte-server/src/main/java/io/airbyte/server/errors/NotFoundExceptionMapper.java:41]: can we write an issue for this and convert this into a TODO with the issue tag so we don't forget?

**Review by davinchia — APPROVED** [review]
One comments. Looks great otherwise! Feel free to merge whenever!

**Pull request merged**
